### PR TITLE
feat: setup experiment to use category service to drive lend by categ…

### PR DESCRIPTION
**Pull request by mcstover**
…ory rows. I'll do the hero section swap in a follow up PR

### CODE REVIEW

File: src/pages/Lend/LendByCategoryPage.vue [hunk 1]
```diff
@@ -199,7 +202,26 @@ export default {
...
+					if (this.categoryServiceExpVersion === 'a') {
+						return category;
+					}
+					// return mapped Category Service category
+					return {
+						id: category?.loanChannelId,
+						name: category?.name ?? '',
+						description: category?.description ?? '',
+						url: category?.url ?? '',
+						loans: category?.savedSearch?.loans ?? []
```
Comment (mcstover) [inline src/pages/Lend/LendByCategoryPage.vue:222]: The loans from the new service are nested under a `savedSearch`. This normalizes them so we don't have to handle that separately in the row presentation components.

File: src/components/LoansByCategory/FeaturedHeroLoanWrapper.vue [hunk 2]
```diff
@@ -36,6 +36,7 @@ import _filter from 'lodash/filter';
 import _get from 'lodash/get';
 
 import featuredLoansQuery from '@/graphql/query/featuredLoansData.graphql';
+// import categoryServiceRowsQuery from '@/graphql/query/lendByCategory/categoryServiceLoanChannels.graphql';
```
Comment (dyersituations) [inline src/components/LoansByCategory/FeaturedHeroLoanWrapper.vue:39]: Remove?

File: src/components/LoansByCategory/FeaturedHeroLoanWrapper.vue [hunk 3]
```diff
@@ -77,6 +78,10 @@ export default {
 			type: Boolean,
 			default: false,
 		},
+		useCategoryService: {
```
Comment (dyersituations) [inline src/components/LoansByCategory/FeaturedHeroLoanWrapper.vue:81]: It looks like this isn't used yet, for follow-up PR?

File: src/components/LoansByCategory/FeaturedHeroLoanWrapper.vue (hunk 2 above)
Comment (mcstover) [reply src/components/LoansByCategory/FeaturedHeroLoanWrapper.vue:39]: Just getting started on the hero section which will also use this query.

File: src/pages/Lend/LendByCategoryPage.vue (hunk 1 above)
Comment (dyersituations) [reply src/pages/Lend/LendByCategoryPage.vue:222]: Yeah I was going to ask about the nesting, it seemed a little confusing.

File: src/pages/Lend/LendByCategoryPage.vue [hunk 4]
```diff
@@ -365,9 +389,14 @@ export default {
...
-						const filteredLoanChannels = loanChannels.filter(loan => {
-							return !ssrLoanIds.includes(loan.id);
+						const loanChannels = this.categoryServiceExpVersion === 'b'
+							? data?.loanCategoriesByLoanChannelIds
+							: _get(data, 'lend.loanChannelsById');
+						const filteredLoanChannels = loanChannels.filter(channel => {
+							return this.categoryServiceExpVersion === 'b'
+								? !ssrLoanIds.includes(channel.loanChannelId)
+								: !ssrLoanIds.includes(channel.id);
+							// return !ssrLoanIds.includes(loan.id);
```
Comment (dyersituations) [inline src/pages/Lend/LendByCategoryPage.vue:399]: Remove?

File: src/pages/Lend/LendByCategoryPage.vue [hunk 5]
```diff
@@ -365,9 +389,14 @@ export default {
 						_each(this.categories, category => {
 							ssrLoanIds.push(category.id);
 						});
-						const loanChannels = _get(data, 'lend.loanChannelsById');
-						const filteredLoanChannels = loanChannels.filter(loan => {
-							return !ssrLoanIds.includes(loan.id);
+						const loanChannels = this.categoryServiceExpVersion === 'b'
```
Comment (dyersituations) [inline src/pages/Lend/LendByCategoryPage.vue:392]: Might be worth putting `this.categoryServiceExpVersion === 'b'` in a computed value. It is used a decent amount, including in the template.

**Review by dyersituations — APPROVED** [review]

File: src/pages/Lend/LendByCategoryPage.vue (hunk 1 above)
Comment (mcstover) [reply src/pages/Lend/LendByCategoryPage.vue:222]: It's definitely different but what's happening is they are automatically mapping a saved search id onto savedSearch so we've added a layer to handle mapping the filters to FLSS in the service...

**Comment by mcstover** [comment]
I'm pushing the clean up items now, but I've discovered these changes altered the lazy loading of the channels so I'll fix that and update this pr early next week.

File: src/pages/Lend/LendByCategoryPage.vue [hunk 6]
```diff
@@ -546,15 +588,18 @@ export default {
...
 								imgRetinaSize: this.showHoverLoanCards ? 'w960h600' : 'w960h720',
 							},
 						}).then(({ data }) => {
-							const fetchedCategory = data?.lend?.loanChannelsById?.[0];
-							if (fetchedCategory?.loans?.values?.length) {
+							const fetchedCategory = this.categoryServiceExpActive
+								? data?.loanCategoriesByLoanChannelIds?.[0]
+								: data?.lend?.loanChannelsById?.[0];
+							if (fetchedCategory?.loans?.values?.length
+								|| fetchedCategory?.savedSearch?.loans?.values?.length) {
```
Comment (mcstover) [inline src/pages/Lend/LendByCategoryPage.vue:602]: Checking our additional savedSearch data structure for loans brought back the lazy loading funcitonality! Previously we were fetching the categories but not rendering them b/c they weren't added to the `realCategories` array.

**Comment by mcstover** [comment]
I'd like to go ahead and merge this as it will provide a verification ground for the category rows. Then I'll follow up with a PR for the HeroLoan query as there will be some exp cleanup tied to that section.

File: src/components/LoansByCategory/FeaturedHeroLoanWrapper.vue [hunk 7]
```diff
@@ -94,24 +95,26 @@ export default {
...
 		preFetch(config, client) {
-			// check if hero channel experiment is enabled
-			return getExperimentSettingAsync(client, heroChannelExpKey)
-				.then(({ enabled }) => {
-					// use ml query to get loan channel to use if experiment is enabled
-					if (enabled) {
-						return getHeroChannelAsync(client);
-					}
-					// otherwise use default query
-					return getFeaturedLoanChannelAsync(client);
```
Comment (mcstover) [inline src/components/LoansByCategory/FeaturedHeroLoanWrapper.vue:105]: Now that the ML served hero channel is the default, we no longer fallback to the Kiva settings options. Rather we will fallback to the hard coded channel if `ml` doesn't resolve one.

File: src/components/LoansByCategory/FeaturedHeroLoanWrapper.vue [hunk 8]
```diff
@@ -148,17 +141,28 @@ export default {
...
 		}
 
-		// get array of loan channels, accounting for different structure of loan channel queries
-		const loanChannels = allLoanData?.lend?.featuredLoanChannel ?? allLoanData?.ml?.getOrderedChannelsByIds ?? [];
+		let loanChannels = [];
+		if (this.useCategoryService) {
+			// get array of loan channels, accounting for different structure of loan channel queries
+			loanChannels = allLoanData?.loanCategoriesByLoanChannelIds ?? [];
+		} else {
+			// get array of loan channels, accounting for different structure of loan channel queries
```
Comment (dyersituations) [inline src/components/LoansByCategory/FeaturedHeroLoanWrapper.vue:149]: Duplicate comment could probably be removed.

**Review by dyersituations — APPROVED** [review]

**Comment by mcstover** [comment]
@emuvente if you have any additions or changes you'd like to me to make, please let me know or post here.

**Pull request merged**